tests/src/whisk/core/controller/test/RejectBigEntitesTests.scala

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.controller.test
+
+import scala.language.postfixOps
+
+import org.junit.runner.RunWith
+import org.scalatest.Finders
+import org.scalatest.junit.JUnitRunner
+
+import spray.http.StatusCodes.RequestEntityTooLarge
+import spray.http.StatusCodes.OK
+import spray.httpx.SprayJsonSupport.sprayJsonMarshaller
+import spray.httpx.SprayJsonSupport.sprayJsonUnmarshaller
+import spray.httpx.marshalling.ToResponseMarshallable.isMarshallable
+import spray.json.DefaultJsonProtocol.RootJsObjectFormat
+import spray.json.pimpString
+import spray.routing.Directive.pimpApply
+import whisk.common.TransactionId
+import whisk.core.controller.EntitySizeChecking
+import whisk.core.entity.LimitedWhiskEntityPut
+import whisk.core.entity.Parameters
+
+/**
+ * Tests Actions API.
```

Comment:
see notes above - the api tests belong in the api tests - you're not really testing the api below.
there could be an independent unit test just for the directive.